lib/galaxy/datatypes/mothur.py

```diff
@@ -0,0 +1,937 @@
+        >>> fname = get_test_fname( 'mothur_datatypetest_true.mothur.quan' )
+        >>> Quantile().sniff( fname )
+        True
+        >>> fname = get_test_fname( 'mothur_datatypetest_false.mothur.quan' )
+        >>> Quantile().sniff( fname )
+        False
+        """
+        headers = get_headers(filename, sep='\t')
+        count = 0
+        for line in headers:
+            if not line[0].startswith('@') and not line[0].startswith('#'):
+                if len(line) != 7:
+                    return False
+                try:
+                    int(line[0])
+                    float(line[1])
+                    float(line[2])
+                    float(line[3])
+                    float(line[4])
+                    float(line[5])
+                    float(line[6])
+                except:
+                    return False
+                count += 1
+        if count > 0:
+            return True
+
+        return False
+
+
+class LaneMask(Text):
+    file_ext = 'mothur.filter'
+
+    def sniff(self, filename):
+        """
+        Determines whether the file is a lane mask filter:  1 line consisting of zeros and ones.
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname( 'mothur_datatypetest_true.mothur.filter' )
+        >>> LaneMask().sniff( fname )
+        True
+        >>> fname = get_test_fname( 'mothur_datatypetest_false.mothur.filter' )
+        >>> LaneMask().sniff( fname )
+        False
+        """
+        headers = get_headers(filename, sep='\t')
+        if len(headers) != 1 or len(headers[0]) != 1:
+            return False
+
+        if not re.match('^[01]+$', headers[0][0]):
+            return False
+
+        return True
+
+
+class CountTable(Tabular):
+    MetadataElement(name="groups", default=[], desc="Group Names", readonly=True, visible=True, no_value=[])
+    file_ext = 'mothur.count_table'
+
+    def __init__(self, **kwd):
+        """
+        http://www.mothur.org/wiki/Count_File
+        A table with first column names and following columns integer counts
+        # Example 1:
+        Representative_Sequence total
+        U68630  1
+        U68595  1
+        U68600  1
+        # Example 2 (with group columns):
+        Representative_Sequence total   forest  pastur
+        U68630  1       1       0
+        U68595  1       1       0
+        U68600  1       1       0
+        U68591  1       1       0
+        U68647  1       0       1
+        """
+        Tabular.__init__(self, **kwd)
+        self.column_names = ['name', 'total']
+
+    def set_meta(self, dataset, overwrite=True, skip=1, max_data_lines=None, **kwd):
+
+        data_lines = 0
+        headers = get_headers(dataset.file_name, sep='\t', count=-1)
+        colnames = headers[0]
+        if len(colnames) > 1:
+            dataset.metadata.columns = len(colnames)
+        if len(colnames) > 2:
+            dataset.metadata.groups = colnames[2:]
+            column_types = ['str']
```

Comment:
If I understand it correclt, if len(colnames) is 2, no column_types will be set?

you could use:
```
dataset.metadata.column_types = ['str'] + (['int'] * ( len(headers[0]) -1))
```
outside of an if statement